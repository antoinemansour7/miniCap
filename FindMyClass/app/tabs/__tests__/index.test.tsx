import * as React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../index'; // Adjust the import path based on the test file's location

jest.mock('@/components/HelloWave', () => {
  return {
    HelloWave: () => <div>Hello Wave</div>,
  };
});

jest.mock('@/components/ParallaxScrollView', () => {
  return ({ children }: { children: React.ReactNode }) => (
    <div className="parallax-scroll-view">{children}</div>
  );
});

jest.mock('@/components/ThemedText', () => {
  return {
    ThemedText: ({ children }: { children: React.ReactNode }) => (
      <span className="themed-text">{children}</span>
    ),
  };
});

jest.mock('@/components/ThemedView', () => {
  return {
    ThemedView: ({ children }: { children: React.ReactNode }) => (
      <div className="themed-view">{children}</div>
    ),
  };
});

it('renders HomeScreen correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});