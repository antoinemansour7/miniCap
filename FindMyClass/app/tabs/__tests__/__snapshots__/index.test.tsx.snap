// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomeScreen correctly 1`] = `
<div
  className="parallax-scroll-view"
>
  <div
    className="themed-view"
  >
    <span
      className="themed-text"
    >
      Welcome!
    </span>
    <div>
      Hello Wave
    </div>
  </div>
  <div
    className="themed-view"
  >
    <span
      className="themed-text"
    >
      Step 1: Try it
    </span>
    <span
      className="themed-text"
    >
      Edit 
      <span
        className="themed-text"
      >
        app/(tabs)/index.tsx
      </span>
       to see changes. Press
       
      <span
        className="themed-text"
      >
        F12
      </span>
       
      to open developer tools.
    </span>
  </div>
  <div
    className="themed-view"
  >
    <span
      className="themed-text"
    >
      Step 2: Explore
    </span>
    <span
      className="themed-text"
    >
      Tap the Explore tab to learn more about what's included in this starter app.
    </span>
  </div>
  <div
    className="themed-view"
  >
    <span
      className="themed-text"
    >
      Step 3: Get a fresh start
    </span>
    <span
      className="themed-text"
    >
      When you're ready, run
       
      <span
        className="themed-text"
      >
        npm run reset-project
      </span>
       to get a fresh
       
      <span
        className="themed-text"
      >
        app
      </span>
       directory. This will move the current
       
      <span
        className="themed-text"
      >
        app
      </span>
       to
       
      <span
        className="themed-text"
      >
        app-example
      </span>
      .
    </span>
  </div>
</div>
`;
